src/main/webapp/app/shared/list/FormListField.tsx

```diff
@@ -0,0 +1,159 @@
+          overflowY: 'scroll',
+          minHeight: '100px',
+          maxHeight: '100px',
+          marginTop: '28px',
+        }}
+      >
+        <div className={styles.main}>
+          {this.props.list.map(item => (
+            <div key={item} className={styles.content_container}>
+              <span
+                className={styles.delete}
+                onClick={() => this.props.onDelete(item)}
+              >
+                <i className="fa fa-times-circle"></i>
+              </span>
+              <span style={{ margin: '5px 10px 5px 5px' }}>{item}</span>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+type IFormListFieldProps = {
+  list: string[];
+  addItem: (item: string) => void;
+  deleteItem: (item: string) => void;
+  labelText: string;
+  placeholder?: string;
+  boldLabeL?: boolean;
+};
+
+@observer
+export class FormListField extends React.Component<IFormListFieldProps> {
+  @observable isInputValid = false;
+  @observable inputText = '';
+  @observable errorMessage = '';
+
+  @autobind
+  listLengthValidation(value: any, context: any, input: any, cb: any) {
+    if (this.props.list.length > 0) {
+      cb(true);
+    } else {
+      cb(false);
+      this.errorMessage = 'List cannot be empty!';
+    }
+  }
+
+  @action.bound
+  domainValidation() {
+    if (this.inputText.length === 0) {
+      this.errorMessage = '';
+      return false;
+    } else if (
+      this.props.list.includes(this.inputText) ||
+      !this.inputText.includes('.')
+    ) {
+      this.errorMessage = 'Domain is not valid';
+      return false;
+    }
+    this.errorMessage = '';
+    return true;
+  }
+
+  @action.bound
+  onEnterKey(event: any) {
+    if (event.which === 13 && this.isInputValid) {
+      this.props.addItem(this.inputText);
+      this.inputText = '';
+      this.isInputValid = this.domainValidation();
+    }
+  }
+
+  render() {
+    return (
+      <AvGroup>
+        <Row>
+          <Col>
+            <label
+              className={this.props.boldLabeL ? 'font-weight-bold' : ''}
+              htmlFor="listItemInput"
+            >
+              {this.props.labelText}
+            </label>
+            <div className="input-group">
+              <AvInput
+                id="listItemInput"
+                name="listItemInput"
+                placeholder={this.props.placeholder || ''}
+                value={this.inputText}
+                onChange={(event: any) => {
+                  this.inputText = event.target.value;
+                  this.isInputValid = this.domainValidation();
+                }}
+                validate={{
+                  customLength: this.listLengthValidation,
+                }}
+                onKeyPress={this.onEnterKey}
+              />
+              <div className="input-group-append">
+                <Button
+                  className={`${styles.add} btn btn-light`}
+                  style={{ zIndex: 0 }}
```

Comment:
Why do we need to specify the zindex?

Reply:
When there is a react-select Select component, the dropdown appears under the button.
![image](https://user-images.githubusercontent.com/59149377/141170081-417bc3fa-8b07-43d9-9dc2-e6c6b126eb4d.png)